src/core/layertree/qgslayertreemodellegendnode.h

```diff
@@ -677,8 +677,10 @@ class CORE_EXPORT QgsRasterSymbolLegendNode : public QgsLayerTreeModelLegendNode
      * \param parent attach a parent QObject to the legend node.
      * \param isCheckable set to TRUE to enable the checkbox for the node (since QGIS 3.18)
      * \param ruleKey optional identifier to allow a unique ID to be assigned to the node by a renderer (since QGIS 3.18)
+     * \param parentRuleKey rule key of parent
```

Comment:
```suggestion
     * \param parentRuleKey rule key of parent  (since QGIS 3.36)
```

Also add explanation of when this should be set